draft-rfcs/const-evaluatable-checked.md

```diff
@@ -0,0 +1,19 @@
+# const evaluatable checked
+
+TODO: All of this :)
+
+## Requirements
+
+For this to work, const operations have to be deterministic and
+must not depend on any external state,
+at least when they are used in the type system.
+
+Using floats during CTFE is fully determinstic. So using
+them inside of the type system is fine. CTFE can however
+produce different results than what would happen on real hardware,
+but this is not a concern for const generics.
```

Comment:
The caveat here is: there are some proposals to make CTFE raise an error when a float operation produces a NaN (this would avoid producing a result different from real hardware). So this could be relevant n the context of ensuring that there are no post-monomorphization errors.